Algorithm.Python/EmaCrossUniverseSelectionAlgorithm.py

```diff
@@ -0,0 +1,104 @@
+# 
+# Licensed under the Apache License, Version 2.0 (the "License"); 
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from clr import AddReference
+AddReference("System")
+AddReference("QuantConnect.Algorithm")
+AddReference("QuantConnect.Indicators")
+AddReference("QuantConnect.Common")
+
+from System import *
+from QuantConnect import *
+from QuantConnect.Data import *
+from QuantConnect.Algorithm import *
+from QuantConnect.Indicators import *
+from System.Collections.Generic import List
+import decimal as d
+
+class EmaCrossUniverseSelectionAlgorithm(QCAlgorithm):
+    '''In this algorithm we demonstrate how to define a universe as a combination of use the coarse fundamental data and fine fundamental data'''
+    def Initialize(self):
+        '''Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.'''
+        
+        self.SetStartDate(2010,01,01)  #Set Start Date
+        self.SetEndDate(2015,01,03)    #Set End Date
+        self.SetCash(100000)           #Set Strategy Cash
+        
+        self.UniverseSettings.Resolution = Resolution.Daily        
+        self.UniverseSettings.Leverage = 2        
+
+        self.coarse_count = 10
+        self.averages = { };
+
+        # this add universe method accepts two parameters:
+        # - coarse selection function: accepts an IEnumerable<CoarseFundamental> and returns an IEnumerable<Symbol>
+        self.AddUniverse(self.CoarseSelectionFunction)
+        
+
+    # sort the data by daily dollar volume and take the top 'NumberOfSymbols'
+    def CoarseSelectionFunction(self, coarse):
+        
+        # We are going to use a dictionary to refer the object that will keep the moving averages 
+        for cf in coarse:
+            if cf.Symbol not in self.averages: 
+                self.averages[cf.Symbol] = SymbolData(cf.Symbol)
+
+            # Updates the SymbolData object with current EOD price
+            avg = self.averages[cf.Symbol]
+            avg.update(cf.EndTime, cf.Price)
+
+        # Filter the values of the dict: we only want up-trending securities 
+        values = filter(lambda x: x.is_uptrend, self.averages.values())
+        
+        # Sorts the values of the dict: we want those with greater difference between the moving averages
+        values.sort(key=lambda x: x.scale, reverse=True)
+
+        # we need to return only the symbol objects
+        list = List[Symbol]()
+        for x in values[:self.coarse_count]:
+            print x.symbol, x.scale
```

Comment:
Hi ! I see you generally use self.Log() & here you used print, should it be self.Log() too?

Reply:
Yes, it should be Log